aws/data_source_aws_pricing_product.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"log"
+
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const (
+	awsPricingTermMatch = "TERM_MATCH"
+)
+
+func dataSourceAwsPricingProduct() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsPricingProductRead,
+		Schema: map[string]*schema.Schema{
+			"service_code": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"filters": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"field": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"result": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsPricingProductRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).pricingconn
+
+	params := &pricing.GetProductsInput{
+		ServiceCode: aws.String(d.Get("service_code").(string)),
+		Filters:     []*pricing.Filter{},
+	}
+
+	filters := d.Get("filters")
+	for _, v := range filters.([]interface{}) {
+		m := v.(map[string]interface{})
+		params.Filters = append(params.Filters, &pricing.Filter{
+			Field: aws.String(m["field"].(string)),
+			Value: aws.String(m["value"].(string)),
+			Type:  aws.String(awsPricingTermMatch),
+		})
+	}
+
+	log.Printf("[DEBUG] Reading pricing of products: %s", params)
+	resp, err := conn.GetProducts(params)
+	if err != nil {
+		return fmt.Errorf("Error reading pricing of products: %s", err)
+	}
+
+	if err = verifyProductsPriceListLength(resp.PriceList); err != nil {
```

Comment:
Minor nitpick: given this function is not being used elsewhere and not explicitly unit tested, we can remove the indirection and just include the logic inline here.